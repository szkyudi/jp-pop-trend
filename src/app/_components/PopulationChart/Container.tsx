import { Presenter } from './Presenter.'
import { usePopulationChart } from './usePopuationChart'
import { Populations } from '@/lib/types/populations'

type Props = {
  populations: Populations
}
export function Container({ populations }: Props) {
  const { dataList } = usePopulationChart(populations)

  return <Presenter dataList={dataList} />
}
