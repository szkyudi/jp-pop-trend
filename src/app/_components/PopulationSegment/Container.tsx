'use client'

import { Presenter } from './Presenter'
import { usePopulationSegment } from './usePopulationSegment'

export function Container() {
  const { segmentItems } = usePopulationSegment()

  return <Presenter items={segmentItems} />
}
