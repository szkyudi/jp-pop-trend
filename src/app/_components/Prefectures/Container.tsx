'use client'

import { PrefecturesPresenter } from './Presenter'
import { components } from '@/lib/api/schema'
import { createContext, useState } from 'react'

export const CheckedPrefecturesContext = createContext<number[]>([])

type Props = {
  prefectures: components['schemas']['Prefecture'][]
  defaultCheckedPrefectures?: number[]
}
export function PrefecturesContainer({
  prefectures,
  defaultCheckedPrefectures = [],
}: Props) {
  const [checkedPrefectures, setCheckedPrefectures] = useState<number[]>(
    defaultCheckedPrefectures,
  )

  const handleChange = (prefCode: number) => (checked: boolean) => {
    if (checked) {
      setCheckedPrefectures([...checkedPrefectures, prefCode])
    } else {
      setCheckedPrefectures(
        checkedPrefectures.filter((code) => code !== prefCode),
      )
    }
  }

  return (
    <CheckedPrefecturesContext.Provider value={checkedPrefectures}>
      <PrefecturesPresenter
        prefectures={prefectures}
        onChange={handleChange}
        checkedPrefectures={checkedPrefectures}
      />
    </CheckedPrefecturesContext.Provider>
  )
}
