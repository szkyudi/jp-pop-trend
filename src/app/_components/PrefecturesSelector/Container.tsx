'use client'

import { Presenter } from './Presenter'
import {
  useCheckedPrefCodes,
  useSetCheckedPrefCodes,
} from '@/app/_contexts/checkedPrefCodes'

import { components } from '@/lib/api/schema'

type Props = {
  prefectures: components['schemas']['Prefecture'][]
}
export function Container({ prefectures }: Props) {
  const checkedPrefectures = useCheckedPrefCodes()
  const setCheckedPrefCodes = useSetCheckedPrefCodes()

  const handleChange = (prefCode: number) => (checked: boolean) => {
    if (checked) {
      setCheckedPrefCodes([...checkedPrefectures, prefCode])
    } else {
      setCheckedPrefCodes(
        checkedPrefectures.filter((code) => code !== prefCode),
      )
    }
  }

  return (
    <Presenter
      prefectures={prefectures}
      onChange={handleChange}
      checkedPrefectures={checkedPrefectures}
    />
  )
}
