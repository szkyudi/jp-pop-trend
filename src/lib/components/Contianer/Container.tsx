import s from './Container.module.scss'
import { ComponentPropsWithoutRef } from 'react'

type Props = ComponentPropsWithoutRef<'div'>
export function Container({ children }: Props) {
  return <div className={s.container}>{children}</div>
}
