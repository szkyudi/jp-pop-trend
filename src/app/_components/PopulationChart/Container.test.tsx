import * as stories from './Container.stories'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/react'

const { Default } = composeStories(stories)

it('表示が変わっていないこと', () => {
  const { container } = render(<Default />)

  expect(container).toMatchSnapshot()
})
