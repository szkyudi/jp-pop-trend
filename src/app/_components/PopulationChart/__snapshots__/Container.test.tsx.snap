// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`表示が変わっていないこと 1`] = `
<div>
  <div
    class="_container_170fcb"
  >
    <div
      class="recharts-responsive-container"
      style="width: 100%; height: 480px; min-width: 720px;"
    />
  </div>
</div>
`;
